themes/main/assets/js/register_index.jsx

```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import Register from './registration/pages/Register';
+
+const container = document.getElementById('root');
+const root = createRoot(container);
+root.render(<Register year={ container.dataset.year } />);
```

Comment:
**Annotation**:
I made the react scripts read from `data-` attributes on the `root` div as a way to pass information in from the page to the react component.  See the shortcode files (`themes/main/layouts/shortcodes/registration_form.html`) to see how those are added in.